### ISSUE TITLE
Content-type header not respecting Accept header for application/scim+json

### ISSUE THREAD

**Issue by wooly** [RIPAGlobal/scimitar#59]
Hi there,

We're trying to use the Microsoft Azure AD SCIM Validator tool [here](https://scimvalidator.microsoft.com/), and we're coming across an issue as shown in the screenshot below:

<img width="1724" alt="MicrosoftTeams-image (13)" src="https://github.com/RIPAGlobal/scimitar/assets/351076/b8cabb9a-e9ff-4b58-8ff2-70f116965e5f">

Testing this in postman, when setting the `Accept` header in the request to the `/Users` endpoint, we get a json response, but the content-type header isn't using `application/scim+json`, which I'm guessing is what the validator is expecting?

<img width="1343" alt="MicrosoftTeams-image (14)" src="https://github.com/RIPAGlobal/scimitar/assets/351076/5da28e70-da75-47ee-903d-10719aa311cd">

Is there anything we can do to set the response content-type header to `application/scim+json` to appease the validator?

Thanks!

**Comment by chaskett** [comment]
@pond  any chance you have any thoughts on this? :-)

**Comment by pond** [comment]
Not much time to look into this, but in the short term, your controllers could add a `before_action` filter that does something like:

```
response.set_header('CONTENT_TYPE', Mime::Type.lookup_by_extension(:scim).to_s)
```

...as a blunt force solution, perhaps evolving that if you wish to detect Accept in Request. It's probably something that Scimitar's base controllers out to do via `render`, if it's not happening already, but a quick glance though specs didn't make that too clear (other than MDN implying that if Accept is present _in general_ then Content-Type _ought_ to respond with the _chosen_ content type - but if you only accept one thing, there's not much of a choice).

**Comment by chaskett** [comment]
Thanks! I will aim to feedback what how we solve this.

**Comment by wooly** [comment]
We've ended up patching the Scimitar application controller's `add_mandatory_response_header` method to add the `CONTENT_TYPE` header like you suggested.

**Comment by kwent** [comment]
Running into the same issue. @wooly could you shared a snippet of code ?

**Comment by wooly** [comment]
@kwent it's a one-liner here: https://github.com/RIPAGlobal/scimitar/commit/e6843599e613112837ee1ca8066eb6f189332c80

**Comment by pond** [comment]
This is (well, should be!) fixed by https://github.com/RIPAGlobal/scimitar/pull/85.

**Comment by pond** [comment]
I've pushed v2.6.1 including #85 now, along with v1.7.1 which back-ports the fix via #86.
